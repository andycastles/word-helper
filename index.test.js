import { describe, expect, test } from 'vitest';
import { execSync } from 'child_process';

describe('wordle utility', () => {
    test('generates error and help without params', () => {
        try {
            const output = execSync('node index.mjs').toString().trim();
            expect(output).toMatchSnapshot();
        } catch (error) {
            expect(error.message).toMatchSnapshot();
        }
    });
    test('correct output with first letter specified', () => {
        try {
            const output = execSync(
                'node index.mjs -1 abcdefghijklmnopqrstuvwxy'
            )
                .toString()
                .trim();
            expect(output).toMatchSnapshot();
        } catch (error) {
            expect(error.message).toMatchSnapshot();
        }
    });
    test('correct output with all letters specified', () => {
        try {
            const output = execSync('node index.mjs shite').toString().trim();
            expect(output).toMatchSnapshot();
        } catch (error) {
            expect(error.message).toMatchSnapshot();
        }
    });
});
