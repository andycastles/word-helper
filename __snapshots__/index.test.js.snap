// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`wordle utility > correct output with all letters specified 1`] = `
"Template:  GREAT
Require:   (no required letters)
Skip:      (no skipped letters)

Possible words:
GREAT"
`;

exports[`wordle utility > correct output with first letter specified 1`] = `
"Template:  __XXX
Require:   (no required letters)
Skip:      (no skipped letters)
Cannot be in position 1:      B C D E F G H I J K L M N O P Q R S T U V W X Y Z

Possible words:
AAXXX ABXXX ACXXX ADXXX AEXXX AFXXX AGXXX AHXXX AIXXX AJXXX AKXXX ALXXX AMXXX ANXXX AOXXX APXXX AQXXX ARXXX ASXXX ATXXX AUXXX AVXXX AWXXX AXXXX AYXXX AZXXX"
`;

exports[`wordle utility > generates error and help without params 1`] = `
"Command failed: node index.mjs
error: missing required argument 'template'

Usage: wordle [options] <template>

Wordle helper utility

Arguments:
  template                 Specify letters in their position or '_' for unknown
                           letter

Options:
  -s, --skip <letters>     Specify which letters cannot be used (default: "")
  -r, --require <letters>  Specify which letters must be used (default: "")
  -w, --word               Only show combinations which are valid words
  -c, --count              Show count of valid words
  -1, --first <letters>    Specify which letters cannot be in the first
                           position (default: "")
  -2, --second <letters>   Specify which letters cannot be in the second
                           position (default: "")
  -3, --third <letters>    Specify which letters cannot be in the third
                           position (default: "")
  -4, --fourth <letters>   Specify which letters cannot be in the fourth
                           position (default: "")
  -5, --fifth <letters>    Specify which letters cannot be in the fifth
                           position (default: "")
  -h, --help               display help for command
"
`;
